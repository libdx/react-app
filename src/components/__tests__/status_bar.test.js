import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import StatusBar from '../status_bar';

describe('StatusBar', () => {
  const title = 'A Title';
  const sortOptions = [
    { id: 'RELEASE_DATE', title: 'release date' },
    { id: 'RATING', title: 'rating' },
  ];
  let element;
  let component;
  let onOptionClick;

  beforeEach(() => {
    onOptionClick = jest.fn();
    element = <StatusBar
            title={title}
            options={sortOptions}
            activeOptionID={sortOptions[0].id}
            onOptionClick={onOptionClick}
        />;
    component = shallow(element);
  });

  it('shows correct title', () => {
    expect(component.text()).toEqual(expect.stringContaining(title));
  });

  it('shows sort options', () => {
    expect(component.text()).toEqual(expect.stringContaining(sortOptions[0].title));
    expect(component.text()).toEqual(expect.stringContaining(sortOptions[1].title));
  });

  it('shows active option', () => {
    const option = component.find('.sort-opt-active');
    expect(option.exists()).toBe(true);
  });

  it('shows inactive option', () => {
    const option = component.find('.sort-opt');
    expect(option.exists()).toBe(true);
  });

  it('passes correct id on click', () => {
    const option = component.find('.sort-opt');
    option.simulate('click');
    expect(onOptionClick).toBeCalledWith(sortOptions[1].id);
  });

  it('renders correctly', () => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
