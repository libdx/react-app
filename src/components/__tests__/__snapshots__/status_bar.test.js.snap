// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar renders correctly 1`] = `
<div
  className="status-bar"
>
  <div
    className="row"
  >
    <span
      className="col-6"
    >
      A Title
    </span>
    <span
      className="col-2"
    >
      Sort by
    </span>
    <span
      className="col-2 sort-opt-active"
      onClick={[Function]}
    >
      release date
    </span>
    <span
      className="col-2 sort-opt"
      onClick={[Function]}
    >
      rating
    </span>
  </div>
</div>
`;
